config/initializers/cache_store.rb

```diff
@@ -0,0 +1,44 @@
+require "active_support"
+require "dalli"
+
+module CacheStore
+  CACHE_EXPIRY = 43_200 # 12 hours
+
+  def self.memcached
+    @memcached ||=
+      Dalli::Client.new(
+        ENV["MEMCACHIER_SERVERS"], {
+          username: ENV["MEMCACHIER_USERNAME"],
+          password: ENV["MEMCACHIER_PASSWORD"],
+          expires_in: CACHE_EXPIRY,
+        }
+      )
+  end
+
+  def self.filestore
+    @filestore ||= ActiveSupport::Cache::FileStore.new(".cache")
+  end
+
+  def self.store
+    if ENV["RACK_ENV"] == "production"
+      Store.new(memcached)
+    else
+      Store.new(filestore)
+    end
+  end
+end
+
+class CacheStore::Store < SimpleDelegator
```

Comment:
Might be worth adding some tests for this file:

1. That `clear` delegates to `flush_all` if available, or `clear` if not
2. That `write` delegates to `set` if available, or `write` if not
3. That `foo` is automatically delegated to the passed object